onnxruntime/core/providers/cpu/nn/string_normalizer.cc

```diff
@@ -8,11 +8,18 @@
 #ifdef _MSC_VER
 #include <codecvt>
 #include <locale.h>
-#elif defined(__APPLE__) || defined(__ANDROID__)
+#elif defined(__ANDROID__)
 #include <codecvt>
 #else
 #include <limits>
 #include <iconv.h>
+
+#if defined(__APPLE__)
+// Specify -liconv here so we don't need to add link flags for libiconv elsewhere, e.g., for iOS apps that statically
+// link to ORT.
+asm(".linker_option \"-liconv\"");
```

Comment:
We need to think it again if the iconv library is from GNU.

Reply:
I see. what's the situation on Linux?

Reply:
iconv is a part of POSIX standard. So every Linux system must provide it. Most Linux systems use glibc as the C Runtime, and glibc provides iconv  functions. GNU's libiconv is another implementation of the iconv functions. Usually people do not need to use libiconv on Linux.  Since every Linux program needs to use glibc, there is no license concern for glibc.

Reply:
See this: https://stackoverflow.com/questions/57734434/libiconv-or-iconv-undefined-symbol-on-mac-osx/57734435

Reply:
according to https://www.gnu.org/software/libiconv/ the libiconv library is licensed under LGPL. is its usage a concern in that case?

Reply:
Probably a question for CELA to be sure.

https://docs.opensource.microsoft.com/legal/cela-guidance/cela-license-specific-guidance/#distribute-lgplv21-code
https://docs.opensource.microsoft.com/legal/cela-guidance/cela-license-specific-guidance/#distribute-lgplv3-code